src/win32.rs

```diff
@@ -7,28 +7,128 @@ use raw_window_handle::Win32WindowHandle;
 
 use std::io;
 use std::mem;
-use std::os::raw::c_int;
+use std::ptr::{self, NonNull};
+use std::slice;
 
 use windows_sys::Win32::Foundation::HWND;
-use windows_sys::Win32::Graphics::Gdi::{
-    GetDC, StretchDIBits, ValidateRect, BITMAPINFOHEADER, BI_BITFIELDS, DIB_RGB_COLORS, HDC,
-    RGBQUAD, SRCCOPY,
+use windows_sys::Win32::Graphics::Gdi;
+
+const ZERO_QUAD: Gdi::RGBQUAD = Gdi::RGBQUAD {
+    rgbBlue: 0,
+    rgbGreen: 0,
+    rgbRed: 0,
+    rgbReserved: 0,
 };
 
+struct Buffer {
+    dc: Gdi::HDC,
+    bitmap: Gdi::HBITMAP,
+    pixels: NonNull<u32>,
+    width: i32,
+    height: i32,
+}
+
+impl Drop for Buffer {
+    fn drop(&mut self) {
+        unsafe {
+            Gdi::DeleteDC(self.dc);
+            Gdi::DeleteObject(self.bitmap);
+        }
+    }
+}
+
+impl Buffer {
+    fn new(window_dc: Gdi::HDC, width: i32, height: i32) -> Self {
+        let dc = unsafe { Gdi::CreateCompatibleDC(window_dc) };
```

Comment:
Is this something we could expect to fail.

Reply:
> ReleaseDC must be called from the same thread that called GetDC

That probably means !Send and !Sync

Reply:
It sounds like it's in principle more nuanced than that, and it's fine to access the DC from another thread, as long as it's only in use by one thread at a time and is released in the thread that created it?

But yeah, probably practically this means making `softbuffer::Surface` use `!Sync` and `!Send`. Do we expect that to be an issue for any uses? Should `softbuffer::Context` be usable in multiple threads? (So you could have a thread per window; but that does mean needing `Arc`/`Mutex` for shared state.)

Apparently `winit::event_loop::EventLoop` is `!Send` and `!Sync`. For similar platform reasons.

It looks like currently on Linux (on this branch) `Surface` is automatically `Send`/`Sync` if I compile only the Wayland backend, and `!Send`/`!Sync` if I compile with `x11`. So we probably want to be explicit and consistent about this.